python/jiminy_py/src/jiminy_py/viewer/panda3d/panda3d_visualizer.py

```diff
@@ -742,6 +784,122 @@ def _adjust_offscreen_window_aspect_ratio(self) -> None:
+        """Add a RGB or depth camera to the scene.
+
+        The user is responsible for managing it, ie set its pose in world, get
+        screenshot from it, and remove it when no longer relevant. Manually
+        added cameras is mainly useful for simulating exteroceptive sensors.
+
+        :param name: Name of the camera to be added.
+        :param is_depthmap: Whether the camera output gathers 3 8-bits integers
+                            RGB channels or 1 32-bits floats depth channel.
+        :param size: Resolution (height and width) in pixel of the image being
+                     captured by the camera.
+        """
+        # TODO: Expose optional parameters to set lens type and properties.
+
+        # Make sure that no camera with the same name already exists
+        if name in self._user_cameras:
+            raise ValueError(
+                "A camera with the same name already exists. Please delete "
+                "it by calling `remove_camera` before adding a new one.")
+
+        # Create new offscreen buffer
+        fbprops = FrameBufferProperties()
+        if is_depthmap:
+            fbprops.set_depth_bits(32)
+            fbprops.set_float_depth(True)
+            fbprops.set_multisamples(0)
+        else:
+            fbprops.set_rgba_bits(8, 8, 8, 8)
+            fbprops.set_float_color(False)
+            fbprops.set_depth_bits(16)
+            fbprops.set_float_depth(False)
+            fbprops.set_multisamples(4)
+        winprops = WindowProperties()
+        winprops.set_size(*size)
+        flags = GraphicsPipe.BF_refuse_window
+        buffer = self.graphicsEngine.make_output(
+            self.pipe, f"user_buffer_{name}", 0, fbprops, winprops, flags,
+            self.win.get_gsg(), self.win)
+        if buffer is None:
+            raise RuntimeError("Faulty graphics pipeline on this machine.")
+        self._user_buffers[name] = buffer
+        self.winList.append(buffer)
+
+        # Disable automatic rendering of the buffer of efficiency
+        buffer.set_one_shot(True)
+
+        # Disable color buffer and enable depth buffer
+        if is_depthmap:
+            buffer.set_clear_color_active(False)
+            buffer.set_clear_depth_active(True)
+
+        # Attach a texture as screenshot requires copying GPU data to RAM
+        tex = Texture(f"user_texture_{name}")
+        if is_depthmap:
+            tex.set_format(Texture.F_depth_component)
+        else:
+            tex.set_format(Texture.F_rgb)
+        buffer.add_render_texture(tex, GraphicsOutput.RTM_copy_ram)
+
+        # Create 3D camera region for the scene.
+        # See official documentation about field of view parameterization:
+        # https://docs.panda3d.org/1.10/python/programming/camera-control/perspective-lenses   # noqa: E501  # pylint: disable=line-too-long
+        lens = PerspectiveLens()
+        if is_depthmap:
+            lens.set_fov(50.0)   # field of view angle [0, 180], 40° by default
```

Comment:
I'm relooking at this. Shouldn't these arguments be part of the method's arguments ?
`add_camera(self,
                   name: str,
                   is_depthmap: bool,
                   window_size: Tuple[int, int],
                   lens_near_far: Tuple[float, float],
                   field_of_view: Union[float, Tuple[float,float]) -> None:`

Reply:
Yes. There are still rough corners that need to be polished. I merged this in a rush because I had to move to something else. But else it should be moved as input arguments.

Reply:
See https://github.com/duburcqa/jiminy/pull/826#discussion_r1773526772